### PR TITLE
Passing a currency as string is deprecated

### PR DESCRIPTION
**Pull request by ossinkine**
See https://github.com/moneyphp/money/pull/397

### CODE REVIEW

**Comment by kucharovic** [comment]
Should I add it to new version (v1.1.0) which will require `moneyphp/money: ^3.1`?

Are you OK with that? (cc @enumag)

**Comment by ossinkine** [comment]
If you will bump `moneyphp/money` version to 3.1 I will remove BC code

**Comment by ossinkine** [comment]
I've bumbed `phpmoney` version and removed BC code. You may add 1.1.0 tag after merge.

File: Form/DataTransformer/MoneyToLocalizedStringTransformer.php [hunk 1]
```diff
@@ -19,20 +21,20 @@
...
 
 	/**
-	 * @param string                 $currencyCode ISO currency code
+	 * @param Currency               $currency
 	 * @param int                    $scale
 	 * @param bool                   $grouping
 	 * @param \Money\Currencies|null $currencies
 	 */
-	public function __construct($currencyCode, $scale, $grouping, Currencies $currencies = null)
+	public function __construct($currency, $scale, $grouping, Currencies $currencies = null)
```
Comment (kucharovic) [inline Form/DataTransformer/MoneyToLocalizedStringTransformer.php]: Please add type hint `Currency` to constructor.

**Comment by ossinkine** [comment]
Maybe you update `composer.json` first and remove support PHP 5.5 and I'll rebase my PR?

**Comment by kucharovic** [comment]
@ossinkine you can rebase and I'll merge it.

**Comment by enumag** [comment]
@kucharovic Sure. 👍

File: Form/DataTransformer/MoneyToLocalizedStringTransformer.php (hunk 1 above)
Comment (ossinkine) [reply Form/DataTransformer/MoneyToLocalizedStringTransformer.php]: Changing public method signature is BC break, so I'll throw deprecation notice if string is passed, you'll add hint when 2.0 is ready

Comment (kucharovic) [reply Form/DataTransformer/MoneyToLocalizedStringTransformer.php]: OK.

**Pull request merged**